Framework/API/src/SpectrumInfo.cpp

```diff
@@ -101,6 +101,27 @@ Kernel::V3D SpectrumInfo::position(const size_t index) const {
   return newPos / static_cast<double>(dets.size());
 }
 
+/// Returns the eFixed value of the spectrum with given index.
+double SpectrumInfo::eFixed(const size_t index) const {
+  double eFixed(0);
+  size_t validEFixedSize(0);
+  const auto &dets = getDetectorVector(index);
+
+  for (const auto &det : dets) {
+    const auto &detIndex = m_detectorInfo.indexOf(det->getID());
+    m_detectorInfo.setCachedDetector(detIndex, det);
+    double detectorEFixed = m_detectorInfo.eFixed(detIndex);
+
+    if (detectorEFixed != 0.0) {
```

Comment:
Can we just use the same averaging logic as above if `m_detectorInfo.eFixed(detIndex);` is going to throw?

I think it would best again to avoid an ambiguous scenario where a detector with an `efixed` has been grouped with one that doesn't have one. It's not clear what to do so I'm tempted to say that the client should deal with that.